src/tooling/grain_doc.md

```diff
@@ -0,0 +1,144 @@
+ *
+ * @since v0.1.0
+ * @history v0.2.0: Argument order changed to data-last
+ */
+export let get = (index, array) => {
+  array[index]
+}
+```
+
+## Usage
+
+Before running `grain doc`, you'll want to add doc block comments to your code. Doc block comments are started with `/**` and ended with `*/`. They must exist on the line above a type declaration or exported value. No docs will be generated for non-exported values so you can keep module internals private.
+
+You can run `grain doc <file|dir> -o <file|dir>` to generate markdown documentation for your code. As of Grain v0.5, the `grain doc` command supports directory input to recursively generate documentation for all `.gr` files.
+
+### Without doc blocks
+
+You can even use `grain doc` without doc blocks in your code. It will still generate simple documentation for data types and exported values, such as displaying their names and type signatures.
+
+## Annotations
+
+### @param
+
+```gr
+/**
+ * @param name: description
+ */
+```
+The `@param` annotation indicates a parameter to the function that the doc block is documenting.
+
+These should be specified in order of the function parameters and the name specified before the colon (`:`) should match the parameter name exactly—while not currently enforced, it will be required for named parameters in the future. The content after the colon is the description of the parameter.
+
+When generating docs, the type of the parameter will be detected by the compiler and inserted into the "Parameters" table.
+
+### @returns
+
+```gr
+/**
+ * @returns description
+ */
+```
+
+The `@returns` annotation indicates the value that is returned by the function that the doc block is documenting.
+
+These must be provided a description of the return value. In the Grain stdlib, we don't use the `@returns` annotation if the function returns `Void`.
+
+When generating docs, the type of the return value will be detected by the compiler and inserted into the "Returns" table.
+
+### @example
+
+```gr
+/**
+ * @example code
+ */
+```
+
+The `@example` annotation can be added to show how the documented code is used. Currently, only single line examples are allowed.
+
+When generating docs, the code snippet will be wrapped in a code block tagged for the `grain` language and inserted into an "Examples" section.
+
+### @since
+
+```gr
+/**
+ * @since version
+ */
+```
+
+The `@since` annotation can be used to specify when the documented code was added.
+
+This must be provided a [Semantic Version](https://semver.org/), optionally prefixed with `v`. For example,`v0.2.0` or `0.2.0` are treated the same.
+
+When generating docs, a `<details>` element titled "Added in X.X.X" will be inserted above the type signature. When using this annotation, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "Added in next" instead of the version.
```

Comment:
```suggestion
When generating docs, a `<details>` element titled "Added in X.X.X" will be inserted above the type signature. When using this attribute, you'll be required to specify the `--current-version=X.X.X` flag to the `grain doc` command. This allows you to specify newer versions in your source code, and the generated docs will show "Added in next" instead of the version.
```